shotover-proxy/src/protocols/cassandra_protocol2.rs

```diff
@@ -40,6 +42,7 @@ pub struct CassandraCodec2 {
     current_frames: Vec<Frame>,
     pk_col_map: HashMap<String, Vec<String>>,
     bypass: bool,
+    last_error: Option<String>,
```

Comment:
What is `last_error` used for? 

I can see it being used in `decode` to catch the case where an error occurs twice. Why isn't it reset to `None` on an Ok response?

Reply:
ooooh, this sort of explanation of how the protocol functions would be very valuable as a comment in the source.